src/injected/shadow-initializer.ts

```diff
@@ -33,7 +33,7 @@ export class ShadowInitializer {
         container.appendChild(styleElement);
     }
 
-    private createShadowHost() {
+    private createShadowHost(): HTMLDivElement {
```

Comment:
>HTMLDivElement [](start = 32, length = 14)

I think it would be a slight improvement for this function to encapsulate the implementation detail of which element type happens to be used for the shadow host container (ie, that this method would do better to return HTMLElement rather than HTMLDivElement)

Reply:
Tbh I don't want to change the implementation of this function in this PR. In my opinion, thats not the meaning or scope of this PR, which is specific to only putting a typedef where it is not. I dont mind looking into changing some other time.

Reply:
I think you're misunderstanding; this wouldn't change the implementation, just the type annotation you've added as part of this PR.

---
In reply to: [267558414](https://github.com/Microsoft/accessibility-insights-web/pull/396#discussion_r267558414) [](ancestors = 267558414)

Reply:
ah, i think `createDivWithId` returns a `div` element and it made sense for me to actually be very specific here that it returns a `div` rather than make it `HTMLElement` which is like returning a parent class than actually be specific. I don't really know or may be I am not able to understand this, how is `HTMLElement` better in this case. Sorry.

Reply:
The responsibility of this function is to create the HTML element that will represent our shadow host. Its consumers don't need to know about exactly what tags/attributes/etc are involved in "being a shadow host element" - that's this function's job to encapsulate. Returning HTMLDivElement leaks an implementation detail unnecessarily

Reply:
But if the responsibility of the function was to create an HTML element, the implementation should have something resembling that, it is specifically creating a `div`. I guess changing the name of the function will make more sense in that scenario. For eg: `createShadowContainer` does exactly that and hence it is returning a `HTMLElement` and not any specific `div` or `anchor` or any other element.

Reply:
Changed